json-rpc/types/src/views.rs

```diff
@@ -339,6 +339,8 @@ pub struct BlockMetadata {
     pub version: u64,
     pub timestamp: u64,
     pub chain_id: u8,
+    pub script_hash_allow_list: Vec<BytesView>,
```

Comment:
In general I'm a little bit hesitant to put those items into `BlockMetadata`, the reason is that it will make its definition start to deviate from the original representation of its Move definition in https://github.com/libra/libra/blob/f5f8fa1769cd3cf04cb669b19cb97cf75540915e/language/stdlib/modules/LibraBlock.move#L10. I feel like there is a need for fetching all runtime critical on-chain configs but squashing those different resources might bring some confusion and will harm the future upgrade path in general.

Reply:
the view here should not be named BlockMetaData, it is not related, maybe we should just rename it to MetadataView instead

Reply:
In addition, the `script_hash_allowlist` is pretty big (34 32-byte hashes and growing), which may bloat this view...

I wonder if it would make sense to expose different structs for fairly static metadata (e.g., `chain_id`, allowlist, set of allowed currencies) and highly dynamic metadata (e.g., version, timestamp)?

Reply:
we return chain_id, version & timestamp in every response body, so technically, there is no need to call get_metadata for these information anymore.
Instead of keeping current get_metadata useless, adding onchain configuration is not bad.

I consider it's metadata because the returned data can be used for setup client. 
So splitting into multiple endpoints may not really helpful for client.

Reply:
If that's the case `ConfigurationView` sounds like a better name to me. The point was that the `BlockMetadata` is a confusing name especially with the data we are trying to put over there.

Reply:
BlockMetadata is hidden to client side, if you feel confusing, I can rename it to something else, like MetadataView.
this is doc for the api: https://github.com/libra/libra/blob/master/json-rpc/docs/method_get_metadata.md

it's blockchain / ledger metadata.

add a new "XxxView" does not mean client side sees same name.
Also, for every server side concept, I don't think client need to understand them, most global static stuff can be inside get metadata response.